bgpd/bgp_updgrp_adv.c

```diff
@@ -749,9 +749,14 @@ void subgroup_default_originate(struct update_subgroup *subgrp, int withdraw)
 
 		/* If the peer is on shared nextwork and we have link-local
 		   nexthop set it. */
-		if (peer->shared_network
-		    && !IN6_IS_ADDR_UNSPECIFIED(&peer->nexthop.v6_local))
-			attr.mp_nexthop_len = BGP_ATTR_NHLEN_IPV6_GLOBAL_AND_LL;
+		if (peer->shared_network &&
+			!IN6_IS_ADDR_UNSPECIFIED(&peer->nexthop.v6_local) &&
+			(!bgp_flag_check(bgp,
+				BGP_FLAG_IPV6_LINK_LOCAL_ONLY) ||
+			!IN6_IS_ADDR_UNSPECIFIED(
+				&peer->nexthop.v6_global)))
+			attr.mp_nexthop_len =
```

Comment:
what if `bgp enable-ipv6-ll-only` set here? mp_nexthop_len remains unset here or I missed something?

Reply:
If you compare the diff, you would find out that this code takes care of the old secerio i.e when this feature is not enabled. So in order to kept the old code as it is when the feature is not enabled these checks are added.